api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirServiceRequestDaoImpl.java

```diff
@@ -89,4 +111,275 @@ private Optional<Criterion> handleDateRange(DateRangeParam dateRangeParam) {
+			return observationQuery;
+		}
+		
+		switch (parameterName) {
+			case Observation.SP_DATE:
+				addSimpleDateSearch(observationQuery, "obsDatetime", parameterValue);
+				break;
+			
+			case Observation.SP_PATIENT:
+			case Observation.SP_SUBJECT:
+				addReferenceSearchByUuid(observationQuery, "person", parameterValue);
+				break;
+			
+			case Observation.SP_VALUE_CONCEPT:
+				addReferenceSearchByUuid(observationQuery, "valueCoded", parameterValue);
+				break;
+			
+			case Observation.SP_VALUE_DATE:
+				addSimpleDateSearch(observationQuery, "valueDatetime", parameterValue);
+				break;
+			
+			case Observation.SP_HAS_MEMBER:
+				DetachedCriteria memberQuery = DetachedCriteria.forClass(Obs.class);
+				memberQuery.setProjection(Projections.property("obsGroup"));
+				
+				if (parameterValue != null) {
+					memberQuery.add(Restrictions.eq("uuid", parameterValue));
+				}
+				
+				observationQuery.add(propertyIn("id", memberQuery));
+				break;
+			
+			case Observation.SP_VALUE_STRING:
+				addSimpleSearch(observationQuery, "valueText", parameterValue);
+				break;
+			
+			case Observation.SP_IDENTIFIER:
+				addSimpleSearch(observationQuery, "uuid", parameterValue);
+				break;
+			
+			case Observation.SP_ENCOUNTER:
+				addReferenceSearchByUuid(observationQuery, "encounter", parameterValue);
+				break;
+			
+			case Observation.SP_CATEGORY:
+				if (parameterValue == null) {
+					observationQuery.add(Restrictions.isNotNull("concept"));
+					break;
+				}
+				
+				// TODO: where are the possible values defined? Are all values in ConceptClass relevant?
+				switch (parameterValue) {
+					case "laboratory":
+						addReferenceSearchByUuid(observationQuery, "concept", ConceptClass.LABSET_UUID);
+						break;
+					
+					default:
+						log.warn(
+						    "Failed to add has constraint for observation category with unknown concept  " + parameterValue);
+						// ensure no entries are found
+						addNoResultsCriteria(observationQuery);
+				}
+				break;
+			
+			case Observation.SP_STATUS:
+				addEnumSearch(observationQuery, "status", Obs.Status.class, parameterValue);
+				break;
+			
+			case Observation.SP_CODE:
+				addReferenceSearchByUuid(observationQuery, "concept", parameterValue);
+				break;
+			
+			case Observation.SP_VALUE_QUANTITY:
+				addSimpleDoubleSearch(observationQuery, "valueNumeric", parameterValue);
+				break;
+			
+			case Observation.SP_PERFORMER:
+				DetachedCriteria encounterProviderQuery = DetachedCriteria.forClass(EncounterProvider.class);
+				encounterProviderQuery.setProjection(Projections.property("encounter"));
+				
+				if (parameterValue != null) {
+					encounterProviderQuery.createAlias("provider", "provider");
+					encounterProviderQuery.add(Restrictions.eq("provider.uuid", parameterValue));
+				}
+				
+				observationQuery.add(propertyIn("encounter", encounterProviderQuery));
+				break;
+			
+			// TODO: add explanation on why these values are not implemented
```

Comment:
i would add this explanation in the method javadoc

Reply:
I was unable to determine _why_ these properties couldn't be represented, but added the information to the javadoc.